src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -36,15 +38,25 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long id) {
-        User user = userService.get(id);
+    public void completeOrder(@RequestParam Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
```

Comment:
let's remove this check in this case and just cast principal to UserDetails
same for all methods where you get Authentication object

Reply:
Ok.